Lib/glyphsLib/builder/glyph.py

```diff
@@ -254,17 +254,23 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph, do_color_layers=True):  # noqa:
     # The width may be taken from another master via the customParameters
     # 'Link Metrics With Master' or 'Link Metrics With First Master'.
     master = self.font.masters[layer.associatedMasterId or layer.layerId]
-    metric_source = master.metricsSource.id
-    metric_layer = self.font.glyphs[glyph.name].layers[metric_source]
-    if metric_layer:
-        width = metric_layer.width
-        if layer.width != width:
-            logger.debug(
-                f"{layer.parent.name}: Applying width from master "
-                f"'{metric_source}': {layer.width} -> {width}"
-            )
+    if (
+        "Link Metrics With Master" in master.customParameters
+        or "Link Metrics With First Master" in master.customParameters
+    ):
```

Comment:
> ```
> # The width may be taken from another master via the customParameters 
> # 'Link Metrics With Master' or 'Link Metrics With First Master'.
> ```
> 
> It says `may`, but then the metrics are copied regardless. I added a condition.

There already is a condition in https://github.com/googlefonts/glyphsLib/blob/main/Lib/glyphsLib/classes.py#L1592-L1616, so you're double checking the existence of the custom parameter now.